diff --git a/App/components/listItems/NotificationListItem.tsx b/App/components/listItems/NotificationListItem.tsx
index 8bdace7c22354cc7a5e7752cd3957e8ac80018ba..34c63aae5acd87005869b6994d93c38185d6f186 100644
--- a/App/components/listItems/NotificationListItem.tsx
+++ b/App/components/listItems/NotificationListItem.tsx
@@ -49,6 +49,8 @@ export interface NotificationListItemProps {
   notificationType: NotificationType
   notification: BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord | CustomNotificationRecord
   customNotification?: CustomNotification
+  openSwipeableId?: string | null
+  onOpenSwipeable?: (id: string | null) => void
 }
 
 type DisplayDetails = {
@@ -83,6 +85,8 @@ const NotificationListItem: React.FC<NotificationListItemProps> = ({
   notificationType,
   notification,
   customNotification,
+  openSwipeableId,
+  onOpenSwipeable,
 }) => {
   const navigation = useNavigation<StackNavigationProp<HomeStackParams>>()
   const [store, dispatch] = useStore()
diff --git a/App/screens/Home.tsx b/App/screens/Home.tsx
index 78cd401bd6b0e8e07d5d266274b0229403827462..fd2bbddef5191a7748f77adcbe6318a9bb4e04a6 100644
--- a/App/screens/Home.tsx
+++ b/App/screens/Home.tsx
@@ -40,6 +40,8 @@ const Home: React.FC<HomeProps> = () => {
   const [showTourPopup, setShowTourPopup] = useState(false)
   const screenIsFocused = useIsFocused()
 
+  const [notificationItemId, setNotificationItemId] = useState<string | null>(null)
+
   const styles = StyleSheet.create({
     flatlist: {
       marginBottom: 35,
@@ -50,27 +52,49 @@ const Home: React.FC<HomeProps> = () => {
     (item: any): React.ReactNode => {
       let component: React.ReactNode
       if (item.type === 'BasicMessageRecord') {
-        component = <NotificationListItem notificationType={NotificationType.BasicMessage} notification={item} />
+        component = (
+          <NotificationListItem
+            openSwipeableId={notificationItemId}
+            onOpenSwipeable={setNotificationItemId}
+            notificationType={NotificationType.BasicMessage}
+            notification={item} />
+        )
       } else if (item.type === 'CredentialRecord') {
         let notificationType = NotificationType.CredentialOffer
         if (item.revocationNotification) {
           notificationType = NotificationType.Revocation
         }
-        component = <NotificationListItem notificationType={notificationType} notification={item} />
+        component = (
+          <NotificationListItem
+            openSwipeableId={notificationItemId}
+            onOpenSwipeable={setNotificationItemId}
+            notificationType={notificationType}
+            notification={item}
+          />
+        )
       } else if (item.type === 'CustomNotification' && customNotification) {
         component = (
           <NotificationListItem
+            openSwipeableId={notificationItemId}
+            onOpenSwipeable={setNotificationItemId}
             notificationType={NotificationType.Custom}
             notification={item}
             customNotification={customNotification}
           />
         )
       } else {
-        component = <NotificationListItem notificationType={NotificationType.ProofRequest} notification={item} />
+        component = (
+          <NotificationListItem
+            openSwipeableId={notificationItemId}
+            onOpenSwipeable={setNotificationItemId}
+            notificationType={NotificationType.ProofRequest}
+            notification={item}
+          />
+        )
       }
       return component
     },
-    [customNotification, NotificationListItem]
+    [customNotification, NotificationListItem, notificationItemId, setNotificationItemId]
   )
 
   useEffect(() => {
diff --git a/App/utils/helpers.ts b/App/utils/helpers.ts
index e190a9a6577209a1c54e0b9d0a91e9eddad58d81..b075459acd78689976fe5c60755766790ad649f0 100644
--- a/App/utils/helpers.ts
+++ b/App/utils/helpers.ts
@@ -56,7 +56,7 @@ import {
   getDescriptorMetadata,
 } from './anonCredsProofRequestMapper'
 import { parseCredDefFromId } from './cred-def'
-import { isOpenIdCredentialOffer, isOpenIdPresentationRequest } from './parsers'
+import { InvitationQrTypes, isOpenIdCredentialOffer, isOpenIdPresentationRequest } from './parsers'
 
 export { parsedCredDefNameFromCredential } from './cred-def'
 
@@ -1092,6 +1092,10 @@ export const connectFromScanOrDeepLink = async (
       return
     }
 
+    if (uri.startsWith(InvitationQrTypes.DIDCOMM) && !(uri.includes('c_i=') || uri.includes('oob=') || uri.includes('oobUrl=') || uri.includes('d_m='))) {
+      uri = uri.replace(InvitationQrTypes.DIDCOMM, InvitationQrTypes.HTTPS)
+    }
+
     const aUrl = processBetaUrlIfRequired(uri)
     const receivedInvitation = await connectFromInvitation(aUrl, agent, implicitInvitations, reuseConnection)
 
